src/test/java/generator/DecisionGeneratorTest.java

```diff
@@ -0,0 +1,72 @@
+        Map<Model.Side, TreeMap<Integer, Lane>> lanesMap = new HashMap<>();
+        TreeMap<Integer, Lane> leftTreeMap = new TreeMap<>();
+        for(int i=1; i<=leftLanes; i++) {
+            leftTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> rightTreeMap = new TreeMap<>();
+        for(int i=1; i<=rightLanes; i++) {
+            rightTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> centerTreeMap = new TreeMap<>();
+        centerTreeMap.put(0, null);
+        lanesMap.put(Model.Side.LEFT, leftTreeMap);
+        lanesMap.put(Model.Side.RIGHT, rightTreeMap);
+        lanesMap.put(Model.Side.CENTER, centerTreeMap);
+        return lanesMap;
+    }
+
+    @Test
+    public void generateActionByDecisionMapTest() {
+        Map<Decision, Action> actionByDecision = new HashMap<>();
+        String baseIRI = "anything";
+        MyFactory factoryMock = mock(MyFactory.class);
+        Change_lane changeLaneMock = mock(Change_lane.class);
+        Model modelMock = mock(Model.class);
+        Scenario scenarioMock = mock(Scenario.class);
+
+        doNothing().when(changeLaneMock).addLane_change_by(anyInt());
+        when(factoryMock.createDecision(anyString())).thenAnswer((Answer<Decision>) invocationOnMock -> mock(Decision.class));
+        when(factoryMock.createFollow(anyString())).thenReturn(mock(Follow.class));
+        when(factoryMock.createStop(anyString())).thenReturn(mock(Stop.class));
+        when(factoryMock.createTurn_left(anyString())).thenReturn(mock(Turn_left.class));
+        when(factoryMock.createTurn_right(anyString())).thenReturn(mock(Turn_right.class));
+        when(factoryMock.createChange_lane(anyString())).thenReturn(changeLaneMock);
+        doNothing().when(scenarioMock).addHas_decision(any());
+        when(modelMock.getScenario()).thenReturn(scenarioMock);
+        doNothing().when(modelMock).setActionByDecision(anyMap());
+
+        Map<Model.Side, TreeMap<Integer, Lane>> lanesMap = getExampleLaneMap(3, 2);
+        when(modelMock.getLanes()).thenReturn(lanesMap);
+
+        DecisionGenerator decisionGenerator = new DecisionGenerator(factoryMock, baseIRI, actionByDecision);
+        decisionGenerator.generate(modelMock);
+
+        List<Class> expectedActionClasses = List.of(Follow.class, Stop.class, Turn_right.class, Turn_left.class,
+                Change_lane.class, Change_lane.class, Change_lane.class, Change_lane.class, Change_lane.class);
+        // iterate over expectedActionClasses and remove 1 matching entry from map
+        for(Class actionClass : expectedActionClasses){
+            for(Iterator<Map.Entry<Decision, Action>> it = actionByDecision.entrySet().iterator(); it.hasNext();){
+                Map.Entry<Decision, Action> entry = it.next();
+                if(actionClass.isInstance(entry.getValue())) {
+                    it.remove();
+                    break;
+                }
+            }
```

Comment:
Tutaj zrób po prostu iterację po akcjach i jakoś Assertami to zrób, np jakieś contains na czymś
Ale mam wrażenie, że ten DecisionGenerator nie ma za ładnego interfejsu

Reply:
Chciałem użyć po prostu contains ale w ten sposób sprawdzam jednocześnie czy jest akcja i czy jest ich odpowiednia ilość. Change_Lane się musi powtarzać x razy i tego się nie da tak prosto sprawdzić.